File: modules/module-core/units/python/python-hello/hello.py
"""Minimal demo for the python_venv yoe class.

Invoked on the target by /usr/bin/python-hello, which runs this script
through the python interpreter inside /usr/lib/python-venvs/python-hello.
The pyfiglet import only resolves because that venv's site-packages is on
the interpreter's sys.path -- a regular `python3 -c 'import pyfiglet'`
from the system shell will fail, which is exactly what makes the venv the
unit of distribution.
"""
import sys
from pyfiglet import Figlet


def main(argv):
    text = " ".join(argv[1:]) if len(argv) > 1 else "Hello, yoe!"
    print(Figlet(font="slant").renderText(text))


if __name__ == "__main__":
    main(sys.argv)

File: modules/module-core/units/nodejs/nodejs-hello/hello.js
// Minimal demo for the nodejs_app yoe class.
//
// Invoked on the target by /usr/bin/nodejs-hello, which runs this script
// through node with the package's bundled node_modules on Node's module
// resolution path. The `require("figlet")` only resolves because that
// node_modules tree lives next to this file under
// /usr/lib/node-apps/nodejs-hello -- a plain `node -e 'require("figlet")'`
// from a shell will fail, which is exactly what makes the app directory
// the unit of distribution.
const figlet = require("figlet");

const argv = process.argv.slice(2);
const text = argv.length > 0 ? argv.join(" ") : "Hello, yoe!";

console.log(figlet.textSync(text, { font: "Slant" }));
